packages/babel-helper-create-class-features-plugin/test/fixtures/typescript/constructor-order-super/output.js

```diff
@@ -0,0 +1,29 @@
+class Foo {}
+class Foo2 extends Foo {
+  constructor(incoming) {}
+}
+class Foo3 extends Foo {
+  constructor(incoming) {
+    super();
+    this.incoming = incoming;
+    babelHelpers.defineProperty(this, "value", this.incoming);
+  }
+}
+class Foo4 extends Foo {
+  constructor(incoming) {
+    if (1) {
+      super();
+      this.incoming = incoming;
+      babelHelpers.defineProperty(this, "value", this.incoming);
+    } else {
+      super();
+      this.incoming = incoming;
+      babelHelpers.defineProperty(this, "value", this.incoming);
+    }
+  }
+}
+class Foo5 extends Foo {
+  constructor(incoming) {
+    var c = a ? (super(), (this.incoming = incoming, babelHelpers.defineProperty(this, "value", this.incoming)), this) : b;
```

Comment:
` var c = a ? (super(), (this.incoming = incoming, babelHelpers.defineProperty(this, "value", this.incoming)), this) : b;`
vs
` var c = a ? (super(), this.incoming = incoming, babelHelpers.defineProperty(this, "value", this.incoming), this) : b;`
There is actually a potential bug here when more than two plugins inject initialization statements.
I tried to fix it, but had some difficulty, so I decided to defer to a future PR, which should affect very few people and introduce no new regressions.

The root of the problem is when we use `insertAfter`, it will insert a `SequenceExpression` even if it is already in a `SequenceExpression`.
I'm not sure if we should modify `insertAfter` or we should handle it manually in the plugin.
I tried to modify `insertAfter`, which caused a lot of CI errors. I tried to fix some of them, but there were still some difficulties.
I hope to be able to look into them after we reach a consensus.

Reply:
Don-t those two examples have the same behavior anyway? One just has unnecessary parentheses.

Reply:
Yes, it works fine when enabling these two plugins. If someone uses three plugins that will inject initialization statements, I think it will stay wrong before this PR.

Reply:
https://github.com/babel/babel/pull/15583/files#diff-8550c50c985de30c5ad82ae395d30aaab13c2d34f641cf3780ef3d4533d376b4R97-R104

Because we are currently only searching for siblings of `super()`.